orbit/pkg/execuser/execuser_linux.go

```diff
@@ -18,9 +19,95 @@ import (
 
 // run uses sudo to run the given path as login user.
 func run(path string, opts eopts) (lastLogs string, err error) {
+	args, err := getUserAndDisplayArgs(path, opts)
+	if err != nil {
+		return "", fmt.Errorf("get args: %w", err)
+	}
+
+	args = append(args,
+		// Append the packaged libayatana-appindicator3 libraries path to LD_LIBRARY_PATH.
+		//
+		// Fleet Desktop doesn't use libayatana-appindicator3 since 1.18.3, but we need to
+		// keep this to support older versions of Fleet Desktop.
+		fmt.Sprintf("LD_LIBRARY_PATH=%s:%s", filepath.Dir(path), os.ExpandEnv("$LD_LIBRARY_PATH")),
+		path,
+	)
+
+	cmd := exec.Command("sudo", args...)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	log.Printf("cmd=%s", cmd.String())
+
+	if err := cmd.Start(); err != nil {
+		return "", fmt.Errorf("open path %q: %w", path, err)
+	}
+	return "", nil
+}
+
+// run uses sudo to run the given path as login user and waits for the process to finish.
+func runWithOutput(path string, opts eopts) (output []byte, exitCode int, err error) {
+	args, err := getUserAndDisplayArgs(path, opts)
+	if err != nil {
+		return nil, 0, fmt.Errorf("get args: %w", err)
+	}
+
+	args = append(args, path)
+
+	if len(opts.args) > 0 {
+		for _, arg := range opts.args {
+			args = append(args, arg[0], arg[1])
+		}
+	}
+
+	cmd := exec.Command("sudo", args...)
+	log.Printf("cmd=%s", cmd.String())
+
+	output, err = cmd.Output()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			exitCode = exitErr.ExitCode()
```

Comment:
Should `exitCode` be set to something like `-1` if `err` is not a `*exec.ExitError`?
If not, worth documenting something around the possible return values.

Reply:
good call, updated to `-1` on non-ExitErr errors